src/Microsoft.AspNet.Authentication/AuthenticationHandler.cs

```diff
@@ -189,12 +189,12 @@ protected Task<AuthenticationTicket> AuthenticateOnceAsync()
             }
         }
 
-        protected abstract Task<AuthenticationTicket> AuthenticateAsync();
+        protected abstract Task<AuthenticationTicket> HandleAuthenticateAsync();
 
         public bool ShouldHandleScheme(string authenticationScheme)
         {
             return string.Equals(BaseOptions.AuthenticationScheme, authenticationScheme, StringComparison.Ordinal) ||
-                (BaseOptions.AutomaticAuthentication && string.IsNullOrWhiteSpace(authenticationScheme));
+                (BaseOptions.AutomaticAuthentication && authenticationScheme == string.Empty);
```

Comment:
Well I don't like treating them differently in general, but since we will be using [NotNull] for authenticationScheme everywhere, I don't care too strongly about this.  You have the tiebreaking vote :)